File: Arithmetic/P36.cpp
#include <iostream>
#include <math.h>

#include <stack>

int main(){
    std::stack <int> s;
    int num;

    std::cout << "Enter the number : ";
    std::cin >> num;

    for(; num % 2 == 0; s.push(2), num /= 2);

    for(int i = 3; i <= sqrt(num); i += 2)
        for(; num % i == 0; s.push(i), num /= i);

    if(num > 2) s.push(num);

    std::cout << "( ";
    int c, count;
    for(c = s.top(), count = 0; !s.empty(); s.pop()){
        if(c != s.top()){
            std::cout << "( " << c << " " << count << " ) ";
            c = s.top();
            count = 1;
        }else{
            count++;
        }
    }

    std::cout << "( " << c << " " << count << " ) )";
    return 0;
}

File: Python/P36.py
from math import sqrt

if __name__ == '__main__' :
    stk = []
    num = int(input("Enter the number : "))
    # First find all 2s
    while num%2 == 0 :
        stk.append(2)

        num /= 2

    # Start from 3 and check for further prime nubers
    i = 3
    while i <= sqrt(num) :
        while num % i == 0 :
            stk.append(i)

            num /= i

        i += 2

    # Store the factor in stk
    if num > 2 : stk.append(int(num))

    # Print it in well formated string.
    print("(", end = " ")

    c = stk[0]
    count  = 0

    for i in stk :
        if c != i :
            print("( {} {} ) ".format(c, count), end = "")
            c = i
            count = 1
        else :
            count += 1

    print("( {} {} ) )".format(c, count))